### ISSUE TITLE
Add a forced compilation target if compile = "original"

### ISSUE THREAD

**Issue by wlandau** [wlandau/stantargets#21]
## Prework

* [x] I understand and agree to the [code of conduct](https://github.com/wlandau/stantargets/blob/main/CODE_OF_CONDUCT.md) and  [contributing guidelines](https://github.com/wlandau/stantargets/blob/main/CONTRIBUTING.md).
* [x] If there is [already a relevant issue](https://github.com/wlandau/stantargets/issues), whether open or closed, comment on the existing thread instead of posting a new issue.
* [x] New features take time and effort to create, and they take even more effort to maintain. So if the purpose of the feature is to resolve a struggle you are encountering personally, please consider first posting a "trouble" or "other" issue so we can discuss your use case and search for existing solutions first.

## Proposal

That should keeps us safe from race conditions in parallel computing use cases where the workers share the same models. The key will be to force compilation to override the decisions that `cmdstanr` makes.

**Comment by wlandau** [comment]
Could create conflict with `cmdstanr`'s current timestamp-based invalidation rule.

**Comment by wlandau** [comment]
In functions like `tar_stan_mcmc()`, we can write:

```r
target_compile <- tar_stan_compile_raw(
  name = name_compile,
  command = quote(._stantargets_file_50e43091),
  quiet = quiet,
  log = log,
  dir = dir,
  include_paths = include_paths,
  cpp_options = cpp_options,
  stanc_options = stanc_options,
  force_recompile = force_recompile,
  error = error,
  memory = memory,
  garbage_collection = garbage_collection,
  deployment = deployment,
  priority = priority,
  resources = resources,
  storage = storage,
  retrieval = retrieval,
  cue = cue
)
```

Then in `tar_stan_target_list()`, we could choose between `target_file` and `target_compile`, depending on the `compile` argument.

**Comment by wlandau** [comment]
Should also redirect stdout to the log in compilation targets.

**Comment by wlandau** [comment]
Wait a minute, I already implemented a `target_compile` in the `*_rep()` functions! All I need to do for this issue is force compile in that target and skip compilation in targets immediately downstream. That way, invalidation happens only based on the hash of the file, not the time stamp.
